Backport Fixes into 0.62-stable

NickGerleman opened the pull request:
Also fixup the 0.62 react-native-init script to look for the preview tag instead of master during PR.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4774)

vmoroz commented:
I wonder why not to back port the change where we remove the extra AddRef/Release from user code in C++ applications.

NickGerleman commented:
> I wonder why not to back port the change where we remove the extra AddRef/Release from user code in C++ applications.

This falls into some informal rules from initial discussion of when we would be okay backporting a breaking change. The gist was "we can make breaking changes for something we just added as part of the preview to fix defects". By that bar, breaking ReactContext changes are fine, since we added them in 0.62, but breaking something people have had since 0.61 that wasn't documented in the first preview isn't. This was something of a balance between letting engineers fix issues before they come out, but protecting users from undocumented breaking changes. 

Hoping to draft some of this up into a formally documented policy soon that we can discuss.
